commons/src/main/java/org/infinispan/commons/event/KVPEvent.java

```diff
@@ -0,0 +1,30 @@
+package org.infinispan.commons.event;
+
+import java.io.Serializable;
+
+public class KVPEvent<K, V> implements Serializable {
```

Comment:
I like the idea of having an easy way to get the value and previous value, but I'm not sure about the naming. First, there's nothing about the class that is an event. It's just simply a wrapper. Also, not sure about the use of acronyms here. Although not the greatest choice of names, I feel happier about `KeyValuePrevious` or `KeyValueWithPrevious`. Also, for adding such classes to commons, we should discuss it in the dev list since we try to minimise what we had, and there are similar concepts available around the code base...